opensrp-eusm/src/main/java/org/smartregister/eusm/helper/AppValidateAssignmentHelper.java

```diff
@@ -15,13 +15,21 @@ public AppValidateAssignmentHelper(SyncUtils syncUtils) {
     }
 
     /**
-     * Validate communes only
+     * Filter out previously saved jurisdictions
+     * from preferences
      *
      * @return
      */
     @Override
     protected Set<String> getExistingJurisdictions() {
-        LocationRepository locationRepository = CoreLibrary.getInstance().context().getLocationRepository();
-        return locationRepository.getAllLocations().stream().filter(location -> location.getProperties().getGeographicLevel() == 3).map(location -> location.getId()).collect(Collectors.toSet());
+        Set<String> regionIds = AppUtils.getLocationLevelFromLocationHierarchy(AppConstants.LocationLevels.REGION_TAG);
+        Set<String> districtIds = AppUtils.getLocationLevelFromLocationHierarchy(AppConstants.LocationLevels.DISTRICT_TAG);
+        Set<String> jurisdictionIds = EusmApplication.getInstance().context().userService().fetchJurisdictionIds();
+
+        // Remove district and region ids from assigned communes
+        jurisdictionIds.removeAll(districtIds);
+        jurisdictionIds.removeAll(regionIds);
+
+        return jurisdictionIds;
```

Comment:
@hamza-vd could you explain why this method is overridden? This will help shed some light on why district and region ids are being filtered out from the resulting set of jurisdictions

Reply:
We add `districts` and `regions` to communes (jurisdictions) that are returned at sign-in. It is required bcz it fetches these locations and saves them in the `Location` table of the app DB. So, to compare old assigned jurisdictions, we have to filter out regions and districts.